File: CS/XtraTabWithButtons/ViewInfo/BaseTabPageViewInfoDescendant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraTab.ViewInfo;
using DevExpress.XtraTab;

namespace XtraTabWithButtons
{
    public class BaseTabPageViewInfoDescendant : BaseTabPageViewInfo
    {
        private TabButtonsPanelDescendant _PagePanel;
        public BaseTabPageViewInfoDescendant(IXtraTabPage page)
            : base(page)
        {
            _PagePanel = new TabButtonsPanelDescendant(ViewInfo, Page as XtraTabPageDescendant);
        }

        public TabButtonsPanelDescendant PagePanel
        {
            get { return _PagePanel; }
        }
    }
}

File: VB/XtraTabWithButtons/ViewInfo/BaseTabPageViewInfoDescendant.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraTab.ViewInfo
Imports DevExpress.XtraTab

Namespace XtraTabWithButtons
	Public Class BaseTabPageViewInfoDescendant
		Inherits BaseTabPageViewInfo
		Private _PagePanel As TabButtonsPanelDescendant
		Public Sub New(ByVal page As IXtraTabPage)
			MyBase.New(page)
			_PagePanel = New TabButtonsPanelDescendant(ViewInfo, TryCast(Me.Page, XtraTabPageDescendant))
		End Sub

		Public ReadOnly Property PagePanel() As TabButtonsPanelDescendant
			Get
				Return _PagePanel
			End Get
		End Property
	End Class
End Namespace
